com.reprezen.swagedit/src/com/reprezen/swagedit/editor/SwaggerHyperlink.java

```diff
@@ -0,0 +1,110 @@
+/*******************************************************************************
+ * Copyright (c) 2016 ModelSolv, Inc. and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    ModelSolv, Inc. - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+package com.reprezen.swagedit.editor;
+
+import org.eclipse.jface.text.IRegion;
+import org.eclipse.jface.text.ITextViewer;
+import org.eclipse.jface.text.hyperlink.IHyperlink;
+
+public class SwaggerHyperlink implements IHyperlink {
+
+	private final String text;
+	private final IRegion region;
+	private final IRegion target;
+	private final ITextViewer viewer;
+
+	public SwaggerHyperlink(String text, ITextViewer viewer, IRegion region, IRegion target) {
+		this.text = text;
+		this.viewer = viewer;
+		this.region = region;
+		this.target = target;
+	}
+
+	@Override
+	public IRegion getHyperlinkRegion() {
+		return region;
+	}
+
+	@Override
+	public String getTypeLabel() {
+		return text;
+	}
+
+	@Override
+	public String getHyperlinkText() {
+		return text;
+	}
+
+	@Override
+	public void open() {
+		if (viewer != null) {
+			viewer.setSelectedRange(target.getOffset(), target.getLength());
+			viewer.revealRange(target.getOffset(), target.getLength());
+		}
+	}
+
+	@Override
+	public String toString() {
+		StringBuilder builder = new StringBuilder();
+		builder.append("SwaggerHyperlink [text=");
+		builder.append(text);
+		builder.append(", region=");
+		builder.append(region);
+		builder.append(", target=");
+		builder.append(target);
+		builder.append("]");
+		return builder.toString();
```

Comment:
toString(), equals() and hashCode() are generated by the IDE, so not sure it's worth to use a library to handle that.